target_chains/solana/cli/src/main.rs

```diff
@@ -410,6 +250,93 @@ pub fn process_post_vaa(
     Ok(vaa_pubkey)
 }
 
+pub fn process_write_encoded_vaa(
+    rpc_client: &RpcClient,
+    vaa: &[u8],
+    wormhole: Pubkey,
+    payer: &Keypair,
+) -> Result<Pubkey> {
+    let encoded_vaa_keypair = Keypair::new();
+    let encoded_vaa_size: usize = vaa.len() + EncodedVaa::VAA_START;
+
+    let create_encoded_vaa = system_instruction::create_account(
+        &payer.pubkey(),
+        &encoded_vaa_keypair.pubkey(),
+        Rent::default().minimum_balance(encoded_vaa_size),
+        encoded_vaa_size as u64,
+        &wormhole,
+    );
+    let init_encoded_vaa_accounts = wormhole_core_bridge_solana::accounts::InitEncodedVaa {
+        write_authority: payer.pubkey(),
+        encoded_vaa:     encoded_vaa_keypair.pubkey(),
+    }
+    .to_account_metas(None);
+
+    let init_encoded_vaa_instruction = Instruction {
+        program_id: wormhole,
+        accounts:   init_encoded_vaa_accounts,
+        data:       wormhole_core_bridge_solana::instruction::InitEncodedVaa.data(),
+    };
+
+    process_transaction(
+        rpc_client,
+        vec![create_encoded_vaa, init_encoded_vaa_instruction],
+        &vec![payer, &encoded_vaa_keypair],
+    )?;
+
+    let write_encoded_vaa_accounts = wormhole_core_bridge_solana::accounts::WriteEncodedVaa {
+        write_authority: payer.pubkey(),
+        draft_vaa:       encoded_vaa_keypair.pubkey(),
+    }
+    .to_account_metas(None);
+
+    let write_encoded_vaa_accounts_instruction = Instruction {
+        program_id: wormhole,
+        accounts:   write_encoded_vaa_accounts,
+        data:       wormhole_core_bridge_solana::instruction::WriteEncodedVaa {
+            args: WriteEncodedVaaArgs {
+                index: 0,
+                data:  vaa.to_vec(),
+            },
+        }
+        .data(),
+    };
+
+    process_transaction(
+        rpc_client,
+        vec![write_encoded_vaa_accounts_instruction],
+        &vec![payer],
+    )?;
+
+    let (header, _): (Header, Body<&RawMessage>) = serde_wormhole::from_slice(vaa).unwrap();
+    let guardian_set = GuardianSet::key(&wormhole, header.guardian_set_index);
+
+    let request_compute_units_instruction: Instruction =
+        ComputeBudgetInstruction::set_compute_unit_limit(400_000);
```

Comment:
The compute units here is quite a bummer and increases the size of the payload, but need with 13 signatures